modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -343,7 +364,7 @@
 
 (defmethod unprepare/unprepare-value [:snowflake OffsetDateTime]
   [_ t]
-  (format "timestamp '%s %s %s'" (t/local-date t) (t/local-time t) (t/zone-offset t)))
+  (format "'%s %s %s'::timestamp_tz" (t/local-date t) (t/local-time t) (t/zone-offset t)))
```

Comment:
The current syntax will return a `timestamp without time zone` in snowflake. 
What we want here is a `timestamp with time zone` for OffsetDatetime.

```clojure
;; previously
(dev/query-jdbc-db
    :snowflake
    ["select timestamp '2004-03-19 10:19:09 +09:00'"])
{:cols [{:name "TIMESTAMP '2004-03-19 10:19:09 +09:00'", :base_type :type/DateTime}],
 :rows [[#t "2004-03-19T10:19:09"]]}

;; new change
(dev/query-jdbc-db
    :snowflake
    ["select '2004-03-19 10:19:09 +09:00'::timestamp_tz"])

{:cols [{:name "'2004-03-19 10:19:09 +09:00'::TIMESTAMP_TZ", :base_type :type/DateTimeWithTZ}],
 :rows [[#t "2004-03-19T10:19:09+09:00"]]}

```